src/syn/io/der.rs

```diff
@@ -145,62 +154,245 @@ impl Reader for DerReader {
         &mut self,
         f: F,
     ) -> Result<S, Self::Error> {
-        unimplemented!()
+        let (class, pc, tag) = self.buffer.read_identifier()?;
+        let length = self.buffer.read_length()?;
+
+        match (&class, &pc, &tag) {
+            (Class::Application, _, _) | (Class::Universal, PC::Constructed, 16) | (Class::Universal, PC::Constructed, 17) => {},
+            _ => return Err(Error::InvalidType)
+        }
+
+        eprintln!("[sequence] Class = {:#?}, PC = {:#?}, Tag = {:#?}, Length = {:#?}", class, pc, tag, length);
+
+        // TODO: Why?!
+        if let Class::Application = class {
+            return self.read_sequence::<C, S, F>(f);
+        }
+
+        let range = match length {
+            Length::Definite(l) => Some(self.buffer.read_position..self.buffer.read_position + l as usize),
+            Length::Indefinite => Some(self.buffer.read_position..self.buffer.write_position),
+            _ => None
+        };
+
+        if let Some(ref range1) = range {
+            if self.buffer.byte_len()*8 < range1.end {
+                return Err(Error::EndOfStream);
+            }
+        }
+
+        f(self)
     }
 
     #[inline]
     fn read_sequence_of<C: sequenceof::Constraint, T: ReadableType>(
         &mut self,
     ) -> Result<Vec<T::Type>, Self::Error> {
-        unimplemented!()
+        let (class, pc, tag) = self.buffer.read_identifier()?;
+        let length = self.buffer.read_length()?;
+
+        eprintln!("[sequence of] Class = {:#?}, PC = {:#?}, Tag = {:#?}, Length = {:#?}", class, pc, tag, length);
+
+        if let Class::Universal = &class {
+            if let 16 = &tag {
+            } else {
+                return Err(Error::InvalidType)
+            }
+        }
```

Comment:
With `asn1rs::model::Type` this can be collapsed to
```rust
if !matches!(tag, Tag::Universal(n) if n != 16) {
    return Err(Error::InvalidType);
}
```
or more preferably
```rust
if Tag::DEFAULT_SEQUENCE != tag {
    return Err(Error::InvalidType);
}
```